prow/hook/server.go

```diff
@@ -135,6 +135,16 @@ func (s *Server) demuxEvent(eventType, eventGUID string, payload []byte, h http.
 		if err := json.Unmarshal(payload, &pe); err != nil {
 			return err
 		}
+		// when a branch is deleted, GitHub sends both a `delete' event
+		// as well as a `push' event updating the branch to an invalid
+		// commit SHA. No consumer of push events is expecting an invalid
+		// commit at the HEAD; we should allow plugins to subscribe to
+		// the `delete' event if they want to handle that case. So, if
+		// we can detect that this is the weird `push' event before the
+		// imminent branch deletion, we don't handle it
+		if pe.Deleted || pe.After == "0000000000000000000000000000000000000000" {
```

Comment:
nit: log something? also this logic should be within events.go?

Reply:
It looks like we're swallowing both the delete (`pe.Deleted == true`) and weird push event (`pe.After = 000...`).

How do I subscribe to a delete event after this change?

Reply:
This is mentioned in the issue body. Steve intends for `delete` events to be handled separately.
I'm not sure how I feel about this. It is probably convenient for most Prow plugins, but it makes `hook`s webhook serving diverge from GitHub.
If someone has a GitHub service that handles `push` events and they configure the service as an external Prow plugin, they may be surprised that some of the events are silently filtered out. I think we should probably respect the GitHub API here and allow plugins to skip deletion events themselves (perhaps provide a helper function).

Reply:
I don't see why a plugin would want to handle the weird push to `nil` ref deletion event instead of the actual `X-GitHub-Event: delete` event? These are separate webhook deliveries. We should canvass our current set of plugins, if none use this weird push event to do something on deletion, we should disallow it. There's a perfectly good and non-weird webhook payload specifically for deletions and as a plugin author I would much rather use that.

Reply:
And to be clear, this filter is ONLY for the `X-GitHub-Event: push` webhook payload

Reply:
My understanding is:
1) Github sends a separate delete event
2) Prow does not currently do anything with the delete event, but it would be easy to do so when we need
3) The intent here is for push events to ignore things that are actually deletes.